### PR TITLE
Need to allow the user to select the number of comments they want displayed

### PR DESCRIPTION
**Pull request by tjncgit**
Having trouble appending user input to query string

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -27,12 +27,31 @@ function addRandomGreeting() {
...
+    console.log(param2);
+
+    return [param1, param2];
+    
+    
+}
+
 function getComments() {
+  var param1 = document.getElementById("param1").value;
+  var param2 = document.getElementById("param2").value;
```
Comment (tjncgit) [inline portfolio/src/main/webapp/script.js:43]: get the values input by the user in the html element

File: portfolio/src/main/webapp/script.js [hunk 2]
```diff
@@ -27,12 +27,31 @@ function addRandomGreeting() {
...
+  
+  console.log(param1);
+  console.log(param2);
+  
+//
   fetch('/data')  // sends a request to /my-data-url
```
Comment (tjncgit) [inline portfolio/src/main/webapp/script.js:49]: Thiis is the url I am trying to append to

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -27,12 +27,31 @@ function addRandomGreeting() {
...
+     param2 = document.getElementById('param2').value;
+    console.log(param1);
+    console.log(param2);
+
+    return [param1, param2];
+    
+    
+}
+
 function getComments() {
```
Comment (tjncgit) [inline portfolio/src/main/webapp/script.js:41]: this getComments() function is called when the index.html page loads

File: portfolio/src/main/webapp/index.html [hunk 4]
```diff
@@ -34,9 +34,13 @@ <h2>Comments</h2>
...
-         <form action="/data" method="post">
-         <input type="text" name="input-comment" >
-         <input type="submit" value="submit"/>
+         <form action="/data" method="POST"  >
+         <label for="input-comments">Enter a comment:</label>
+         <input type="text" name="input-comment" id="param2" value="enter comment">
+        <br/>
+         <label for="max-comments">Max number of comments:</label> 
+          <input type="number" name="max-comments" id="param1" min="0" value="0">
+         <input type="submit"  value="submit" >
```
Comment (tjncgit) [inline portfolio/src/main/webapp/index.html:43]: user input takes place in this form element

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -6,7 +6,7 @@
     <link rel="stylesheet" href="style.css">
     <script src="script.js"></script>
   </head>
-  <body onload="getComments()">
+  <body onload="getComments()" >
```
Comment (tjncgit) [inline portfolio/src/main/webapp/index.html:9]: get Comments

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -33,45 +34,62 @@
...
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
     ArrayList<String> comments = new ArrayList<String>();
-    for (Entity commentEntity : results.asIterable()) {
+    for (Entity commentEntity : results.asIterable())
+     {
       String comment = (String) commentEntity.getProperty("comment");
       comments.add(comment);
     }
```
Comment (ohfunc) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:56]: Nit: brackets aren't indented the same

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -33,45 +34,62 @@
...
-
     // Redirect back to the HTML page.
+    response.getWriter().println(max);
     response.sendRedirect("/index.html");
   }
 
+  private int maxComments(HttpServletRequest request) throws IOException {
+      String maxCommentsString = getParameter(request, "max-comments","");
+      int maxComments;
+      try {
```
Comment (ohfunc) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:82]: nit: indentation

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 8]
```diff
@@ -33,45 +34,62 @@
...
+       maxComments = Integer.parseInt(maxCommentsString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + maxCommentsString);
```
Comment (ohfunc) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:85]: Should you log the error (e) as well?

File: portfolio/src/main/webapp/script.js [hunk 9]
```diff
@@ -27,12 +27,31 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
+function getParameters(){
+    let param1 = document.getElementById('param1').value,
+     param2 = document.getElementById('param2').value;
+    console.log(param1);
+    console.log(param2);
```
Comment (ohfunc) [inline portfolio/src/main/webapp/script.js:34]: nit: remove log lines

File: portfolio/src/main/webapp/script.js [hunk 10]
```diff
@@ -27,12 +27,31 @@ function addRandomGreeting() {
...
+
+    return [param1, param2];
+    
+
```
Comment (ohfunc) [inline portfolio/src/main/webapp/script.js:38]: nit: extra whitespace

File: portfolio/src/main/webapp/script.js [hunk 11]
```diff
@@ -27,12 +27,31 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
+function getParameters(){
```
Comment (ohfunc) [inline portfolio/src/main/webapp/script.js:30]: I probably missed it somewhere, but what's the purpose of this function?

File: portfolio/src/main/webapp/script.js [hunk 12]
```diff
@@ -27,12 +27,31 @@ function addRandomGreeting() {
...
+}
+
 function getComments() {
+  var param1 = document.getElementById("param1").value;
+  var param2 = document.getElementById("param2").value;
+  
+  console.log(param1);
+  console.log(param2);
+  
+//
```
Comment (ohfunc) [inline portfolio/src/main/webapp/script.js:48]: Remove log lines and empty comment.

**Pull request merged**